substrate/frame/staking/src/migrations.rs

```diff
@@ -60,12 +61,253 @@ impl Default for ObsoleteReleases {
 #[storage_alias]
 type StorageVersion<T: Config> = StorageValue<Pallet<T>, ObsoleteReleases, ValueQuery>;
 
+/// Migrating all non-slashing related unbounded storage items to bounded
+pub mod v17 {
+	use super::*;
+
+	pub struct VersionUncheckedMigrateV16ToV17<T>(core::marker::PhantomData<T>);
+	impl<T: Config> UncheckedOnRuntimeUpgrade for VersionUncheckedMigrateV16ToV17<T> {
+		fn on_runtime_upgrade() -> Weight {
+			let mut migration_errors = false;
+
+			v16::MaxValidatorsCount::<T>::kill();
+
+			let mut eras_stakers_keys =
+				v16::ErasStakers::<T>::iter_keys().map(|(k1, _k2)| k1).collect::<Vec<_>>();
+			eras_stakers_keys.dedup();
+			for k in eras_stakers_keys {
+				let mut removal_result =
+					v16::ErasStakers::<T>::clear_prefix(k, u32::max_value(), None);
+				while let Some(next_cursor) = removal_result.maybe_cursor {
+					removal_result = v16::ErasStakers::<T>::clear_prefix(
+						k,
+						u32::max_value(),
+						Some(&next_cursor[..]),
+					);
+				}
+			}
```

Comment:
This seems to try to remove all keys in one time, if we don't need multi-block migration and we are sure it is ok then we can do:
```suggestion
			v16::ErasStakers::<T>::clear(u32::max_value(), None);
```

Reply:
We can be sure for Polkadot or Kusama but how can we be sure for every other chain? Maybe there are some limits I'm not aware of

Reply:
These storage items are already empty in Polkadot and Kusama (you can validate in an UI), and this pallet is not parachian-ready, so it should be fine.